node/grpc/server_v2.go

```diff
@@ -140,6 +138,15 @@ func (s *ServerV2) StoreChunks(ctx context.Context, in *pb.StoreChunksRequest) (
 		if err != nil {
 			return nil, api.NewErrorInvalidArg(fmt.Sprintf("failed to verify request: %v", err))
 		}
+		// TODO: move to blob authenticator later
+		for _, blob := range batch.BlobCertificates {
+			if meterer.IsOnDemandPayment(&blob.BlobHeader.PaymentMetadata) {
+				// Batch contains on-demand payments, so the chunk must be from EigenLabsDisperser
+				if in.DisperserID != api.EigenLabsDisperserID {
+					return nil, api.NewErrorInvalidArg("on-demand payments are only allowed for EigenLabsDisperser")
```

Comment:
would be helpful to pinpoint the blob that's on-demand but via a non-eigen disperser in error log

Reply:
yes! would be. I will add